gym/envs/toy_text/guessing_game.py

```diff
@@ -0,0 +1,91 @@
+import gym
+from gym import spaces
+from gym.utils import seeding
+import numpy as np
+
+
+class GuessingGame(gym.Env):
+    """Number guessing game
+
+    The object of the game is to guess within 0.1% of the randomly chosen number
+    within 200 time steps
+
+    After each step the agent is provided with one of four possible observations
+    which indicate where the guess is in relation to the randomly chosen number
+
+    0 - No guess yet submitted (only after reset)
+    1 - Guess is lower than the target
+    2 - Guess is equal to the target
+    3 - Guess is higher than the target
+
+    The rewards are:
+    0 if the agent's guess is outside of 0.1% of the target
+    1 if the agent's guess is inside 0.1% of the target
+
+    The episode terminates after the agent guesses within 0.1% of the target or
+    200 steps have been taken
+
+    The agent will need to use a memory of previously submitted actions and observations
+    in order to efficiently explore the available actions
+
+    The purpose is to have agents optimise their exploration parameters (e.g. how far to
+    explore from previous actions) based on previous experience. Because the goal changes
+    each episode a state-value or action-value function isn't able to provide any additional
+    benefit apart from being able to tell whether to increase or decrease the next guess.
+
+    The perfect agent would likely learn the bounds of the action space (without referring
+    to them explicitly) and then follow binary tree style exploration towards to goal number
```

Comment:
As currently implemented, agents can't 'learn the bounds of the action space' because you `assert self.action_space.contains(action)`.